plugins/woocommerce/src/Internal/Admin/WCAdminSharedSettings.php

```diff
@@ -63,9 +63,42 @@ public function on_woocommerce_blocks_loaded() {
 			\Automattic\WooCommerce\Blocks\Package::container()->get( \Automattic\WooCommerce\Blocks\Assets\AssetDataRegistry::class )->add(
 				$this->settings_prefix,
 				function () {
+					/**
+					 * Filters the shared settings that are passed to the client.
+					 *
+					 * @since 6.4.0
+					 */
 					return apply_filters( 'woocommerce_admin_shared_settings', array() );
 				}
 			);
+			add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_deprecation_scripts' ) );
 		}
 	}
+
+	/**
+	 * Return the deprecated wcSettings properties.
+	 *
+	 * Set value to null to use the default message or provide a custom message.
+	 *
+	 * Nested properties can be defined using dot notation.
+	 *
+	 * @return array
+	 */
+	protected function get_deprecated_properties() {
+		return array(
+			'admin.onboarding' => null,
+		);
+	}
```

Comment:
Given that these deprecations are primarily for frontend JavaScript code, I think we could define these directly in JavaScript so we don't need PHP-JS data transfer, which reduces server response size (though minimal) but also simplifies the logic.

Reply:
@chihsuan Good call! I'll make the change.

Reply:
Hey @chihsuan ✋ 

I've updated the code to define the property list in JS.

Could you review the code one more time please? I've moved things around for reusuability.

1. Added a generic proxy function in https://github.com/woocommerce/woocommerce/pull/55707/files#diff-916fa93bf98ff7dcb904fbda23451f2ab92b723a8e412beff66889abdd09fc99
2. I'm now using it in both [admin-settings.js](https://github.com/woocommerce/woocommerce/pull/55707/files#diff-b5ccd801e97e593229d243c9c5721c56d282ac1e2867df738e62d4ea105bfcbfR75) and [wcsetting-deprecation](https://github.com/woocommerce/woocommerce/pull/55707/files#diff-1e593140447e9917b857820396a32766b99f8286798ed6c0067f3e7ecc161997)

Updated tests accordingly as well.

Thanks!